fixes #2317 - VM's with the name foreman_########

isratrade opened the pull request.

Commit:
fixes #2317 - VM's with the name foreman_########

isratrade commented:
def vm_instance_defaults had "name" (string) as the key and def new_vm merges this with a hash of symbol keys

``` ruby
vm_instance_defaults.merge(attr.to_hash.symbolize_keys)
```

I'm assuming it caused this.

``` ruby
{"name" => foreman_234234234, :name => 'the_real_name'}
```

now vm_instance_defaults has :name as the key, so it will be overwritten by {:name => 'the_real_name'}

domcleal commented:
This looks good.  Should we standardise the vm_instance_defaults a little more and remove :name from EC2, Rackspace and OpenStack, then have them merge with `super` (like VMware does)?

isratrade commented:
@domcleal, I agree with your suggestion and updated the PR.  I also switched the order so it's super.merge(...).  This way, the STI model can overwrite the parent, and not vice versa.

domcleal commented on lib/foreman/model/openstack.rb:
```diff
@@ -97,10 +97,10 @@ def destroy_key_pair
     def vm_instance_defaults
-      {
+      super.merge(
         :name      => "foreman-#{Foreman.uuid}",
```
Can we safely remove these now?

isratrade replied:
Yes, I thought similarly. I will do this.

isratrade replied:
@domcleal, I assume there is no difference between foreman_#{Time.now.to_i} and foreman-#{Foreman.uuid}, right?  just a unique string.

domcleal replied:
Not that I can see.. I would think in both cases the name will get overridden with the FQDN of the host.  Might be worth double checking, I don't use these providers myself.

isratrade replied:
@ohadlevy, ^^^^

ohadlevy commented on app/models/compute_resource.rb:
```diff
@@ -145,7 +145,7 @@ def provider=(value)
   def vm_instance_defaults
     {
-      'name' => "foreman_#{Time.now.to_i}",
+      :name => "foreman_#{Time.now.to_i}"
```
I wonder if we can just change it to HashWithInDifferenceAcesss, it might save more trouble later on?

isratrade commented:
re-committed with HashWithInDifferenceAcesss

isratrade commented:
re-committed again with removing, name      => "foreman-#{Foreman.uuid}

isratrade commented on lib/foreman/model/rackspace.rb:
```diff
@@ -99,10 +99,11 @@ def client
-        :flavor_id => 1, #256 server
-        :name      => "foreman-#{Foreman.uuid}",
-      }
+      #256 server
```
sometimes inline comments can mess things up with the ruby parser.  I tried this in the rails console and it gave an error.

domcleal commented:
:+1: tested with libvirt only.  @ohadlevy, can you review too please?

isratrade commented:
ping?

domcleal commented:
Merged as 557b854, thanks @isratrade!
